src/components/bottom_bar/bottom_bar.tsx

```diff
@@ -78,10 +78,30 @@ export class EuiBottomBar extends Component<Props> {
     }
   }
 
-  componentWillUnmount() {
-    if (this.props.affordForDisplacement) {
-      document.body.style.paddingBottom = '';
+  componentDidUpdate(prevProps: Props) {
+    if (prevProps.affordForDisplacement !== this.props.affordForDisplacement) {
+      if (this.props.affordForDisplacement) {
+        // start affording for displacement
+        const height = this.bar ? this.bar.clientHeight : -1;
+        document.body.style.paddingBottom = `${height}px`;
+      } else {
+        // stop affording for displacement
+        document.body.style.paddingBottom = '';
+      }
+    }
+
+    if (prevProps.bodyClassName !== this.props.bodyClassName) {
+      if (prevProps.bodyClassName) {
+        document.body.classList.remove(prevProps.bodyClassName);
+      }
+      if (this.props.bodyClassName) {
+        document.body.classList.add(this.props.bodyClassName);
+      }
     }
+  }
+
+  componentWillUnmount() {
+    document.body.style.paddingBottom = '';
```

Comment:
Can this be under an if for if `this.props.affordForDisplacement` is true so **in case** the consumer had set it to false because they have a special padding they're adding to the body element they don't want ovreridden.

Reply:
Good thinking, done!

Reply:
Thanks!!!